std/utf.d

```diff
@@ -1078,7 +1079,7 @@ body
     if (str[index] < codeUnitLimit!S)
         return str[index++];
     else
-        return decodeImpl!(true, useReplacementDchar)(str, index);
+        return decodeImpl!(true, cast(DecodeType) useReplacementDchar)(str, index);
```

Comment:
Because the external API uses UseReplacementDchar, whereas `decodeImpl` uses `DecodeType` to differentiate between the 3 cases for decoding (use replacement dchar, throw exception, just return whether decoding is successful). The values line up so that a cast can be used rather than having to check the value and adding overhead.